glib/FastActivator.cs

```diff
@@ -40,6 +40,26 @@ static FastCreateObject FastCtor (Type t, Dictionary<Type, FastCreateObject> cac
 			return method;
 		}
 
+		delegate object FastCreateBoxed (IntPtr ptr);
+		static FastCreateBoxed FastBoxed (Type t, Dictionary<Type, FastCreateBoxed> cache)
+		{
+			FastCreateBoxed method;
+			lock (cache) {
+				if (!cache.TryGetValue (t, out method)) {
+					var newMethod = t.GetMethod ("New", BindingFlags.Static | BindingFlags.Public | BindingFlags.FlattenHierarchy);
+					if (newMethod != null) {
+						var param = Expression.Parameter (typeof (IntPtr));
+						var call = Expression.Call (newMethod, param);
+						var callWithConvert = Expression.Convert (call, typeof (object));
+						cache [t] = method = (FastCreateBoxed)Expression.Lambda (typeof(FastCreateBoxed), callWithConvert, param).Compile ();
```

Comment:
Yeah, we don't get this specific delegate type if we don't pass in the type. Above lambdas do the same.

```
System.InvalidCastException: Specified cast is not valid.
  at GLib.FastActivator.FastBoxed (System.Type t, System.Collections.Generic.Dictionary`2[TKey,TValue] cache) [0x00099] in /Users/therzok/Work/md/gtk-sharp/glib/FastActivator.cs:54
  at GLib.FastActivator.CreateBoxed (System.IntPtr o, System.Type type) [0x00007] in /Users/therzok/Work/md/gtk-sharp/glib/FastActivator.cs:84
  at GLib.Value.ToBoxed () [0x00072] in /Users/therzok/Work/md/gtk-sharp/glib/Value.cs:348
  at GLib.Value.get_Val () [0x0032e] in /Users/therzok/Work/md/gtk-sharp/glib/Value.cs:386
```